lib/value-pairs/cmdline.c

```diff
@@ -340,8 +340,29 @@ vp_cmdline_parse_rekey_shift (const gchar *option_name, const gchar *value,
       return FALSE;
     }
 
-  value_pairs_transform_set_add_func
-  (vpts, value_pairs_new_transform_shift (atoi (value)));
+  value_pairs_transform_set_add_func(vpts,
+                                     value_pairs_new_transform_shift (atoi (value)));
+  return TRUE;
+}
+
+static gboolean
+vp_cmdline_parse_rekey_shift_levels (const gchar *option_name, const gchar *value,
+                                     gpointer data, GError **error)
+{
+  gpointer *args = (gpointer *) data;
+  ValuePairsTransformSet *vpts = (ValuePairsTransformSet *) args[2];
+  gchar *key = (gchar *) args[3];
+
+  vpts = vp_cmdline_rekey_verify (key, vpts, data);
+  if (!vpts)
+    {
+      g_set_error (error, G_OPTION_ERROR, G_OPTION_ERROR_FAILED,
+                   "Error parsing value-pairs: --shift-levels used without --key or --rekey");
+      return FALSE;
+    }
+
+  value_pairs_transform_set_add_func(vpts,
+                                     value_pairs_new_transform_shift_levels (atoi (value)));
```

Comment:
We should exclude zero or less for atoi(value).

Reply:
resolved with using positive_integer in the grammar.

Reply:
wait, this is the cmdline parser...

Reply:
I've added validation and error handling.